### PR TITLE
[Az.Migrate] New cmdlet, GA fixes

### PR DESCRIPTION
**Pull request by prsadhu-ms-idc**
## Description

1. Incorporated Initialize-AzMigrateReplicationInfrastructure as a cmdlet in the Az.Migrate module, from the external script that is run currently today.
2. Made some parameters of New-AzMigrateServerReplication, New-AzMigrateDiskMapping case insensitive.
3. Added support for scale appliance change, to handle new V3 keys.
4. Update documents, tests.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Migrate/custom/New-AzMigrateDiskMapping.ps1 [hunk 1]
```diff
@@ -51,7 +51,25 @@ function New-AzMigrateDiskMapping {
     )
     
     process {
-    
+        # Validate the inputs
```
Comment (msJinLei) [inline src/Migrate/custom/New-AzMigrateDiskMapping.ps1:54]: please use the following style of “if else"
```powershell
if (something) {
}
elseif (something) 
```
Please take care of the space between "if" and "(" or ")" and "{".

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Migrate/custom/New-AzMigrateServerReplication.ps1 [hunk 2]
```diff
@@ -227,6 +228,38 @@ function New-AzMigrateServerReplication {
 
             $null = $PSBoundParameters.Remove('MachineId')
             $null = $PSBoundParameters.Remove('InputObject')
+
+            # Validate some input.
+            if ($LicenseType -eq "NotSpecified"){
```
Comment (msJinLei) [inline src/Migrate/custom/New-AzMigrateServerReplication.ps1:233]: please use the following style of “if else"
```powershell
if (something) {
}
elseif (something) 
```
Please take care of the space between "if" and "(" or ")" and "{".

File: src/Migrate/custom/New-AzMigrateDiskMapping.ps1 (hunk 1 above)
Comment (msJinLei) [reply src/Migrate/custom/New-AzMigrateDiskMapping.ps1:54]: You can use validate set to validate DiskType

Comment (prsadhu-ms-idc) [reply src/Migrate/custom/New-AzMigrateDiskMapping.ps1:54]: I have removed the if-else blocks, and used ValidateSet.

File: src/Migrate/custom/New-AzMigrateServerReplication.ps1 (hunk 2 above)
Comment (prsadhu-ms-idc) [reply src/Migrate/custom/New-AzMigrateServerReplication.ps1:233]: I  have removed the if-else blocks, and used ValidateSet.

Comment (prsadhu-ms-idc) [reply src/Migrate/custom/New-AzMigrateServerReplication.ps1:233]: I have formatted the other custom ps1 files as well.

File: src/Migrate/custom/New-AzMigrateDiskMapping.ps1 [hunk 3]
```diff
@@ -32,14 +32,18 @@ function New-AzMigrateDiskMapping {
...
         [System.String]
         # Specifies whether the disk contains the Operating System for the source server to be migrated.
         ${IsOSDisk},
 
         [Parameter(Mandatory)]
+        [ValidateSet("Standard_LRS", "Premium_LRS", "StandardSSD_LRS")]
+        [ArgumentCompleter( { "Standard_LRS", "Premium_LRS", "StandardSSD_LRS" })]
         [Microsoft.Azure.PowerShell.Cmdlets.Migrate.Category('Path')]
-        [Microsoft.Azure.PowerShell.Cmdlets.Migrate.Support.DiskAccountType]
+        [System.String]
```
Comment (msJinLei) [inline src/Migrate/custom/New-AzMigrateDiskMapping.ps1:46]: You don't need to change enum type to string. To add a validate set is enough. please refer to the following codes
https://github.com/Azure/azure-powershell/blob/8cd4f3a23f5b9db6dfbe7bcdb35964e4528872a8/src/MySql/custom/New-AzMySqlServer.ps1#L88

File: src/Migrate/custom/New-AzMigrateDiskMapping.ps1 [hunk 4]
```diff
@@ -51,12 +55,22 @@ function New-AzMigrateDiskMapping {
...
-        $DiskObject.DiskType = $DiskType
-        $DiskObject.IsOSDisk = $IsOSDisk
-        if($PSBoundParameters.ContainsKey('DiskEncryptionSetID')){
+
+        $validDiskTypeSpellings = @{ 
+            Standard_LRS    = "Standard_LRS";
+            Premium_LRS     = "Premium_LRS";
+            StandardSSD_LRS = "StandardSSD_LRS"
+        }
+        $DiskObject.DiskType = $validDiskTypeSpellings[$DiskType]
```
Comment (msJinLei) [inline src/Migrate/custom/New-AzMigrateDiskMapping.ps1:66]: The map can be removed

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Review by msJinLei — APPROVED** [review]

**Pull request merged**